designs/panorama_plugin/panorama_plugin.md

```diff
@@ -0,0 +1,65 @@
+# Panorama Security Plugin
+This is a security appliance plugin for the Panorama Security Management Appliance, which orchestrates PaloAlto virtual firewalls deployed by OSC. The plugin shall conform to the [security manager api](https://github.com/opensecuritycontroller/security-mgr-api) specification. We propose to have:
+- A Panorama Dynamic address group for each OSC security policy. It will be labeled by a tag and targeted by a _Panorama_ security policy.
+- In Panorama, we define two classes of tags: one to associate Panorama Address Objects with a security group, and another -- with a security policy.
+- The security group tags are managed by osc (through the Panorama API) and the policy tags are managed manually on Panorama.
+
+## Assignees
+Dmitry Gerenrot ([dmitryintel](https://github.com/dmitryintel)).
+
+## Constraints and Assumptions
+The Panorama version is at least 7.1. The firewalls are running PAN-VM-KVM-7.1.4.
+
+## Design Changes
+
+We propose the following workflow for creating a security group from OSC.
+
+- Begin by creating a distributed appliance on OSC with PAN-VM-KVM-7.1.4 as the service function. OSC then creates a Device Group under panorama via the Panorama API. (For example, **DistributedAppGroup**).
+- Proceed by defining the security policies on panorama as follows:
+	- For each policy to be managed by the **Panorama Security Plugin**, there will be a dynamic address group (DAG) under the **DistributedAppGroup** and a _shared_ tag, e.g., **OSC_Policy_1**. Panorama administrators _create these manually_.
```

Comment:
Each policy rule on panorama can reference DAG's under  the **DistributedAppGroup**. The DAG's are constructed manually by the security/panorama administrators by using shared tags for their match criteria for example